### ISSUE TITLE
issue with installing on RPi3

### ISSUE THREAD

**Issue by catalyst1987** [huginn/huginn#2199]
Hello,
    I'm currently trying to install Huginn to a RPi3 that has Ubuntu Mate 17.04.  I am having difficulties at this part:

sudo bundle exec rake production:export --trace
** Invoke production:export (first_time)
** Invoke production:check (first_time)
** Execute production:check
** Execute production:export
** Execute production:stop
Stopping huginn ...
Exporting new services ...

It seems to remain at "Exporting new services" and never seems to complete.  Was wondering if this is a known problem or if I have configured something wrong?  Not really sure how to troubleshoot this.

Many Thanks for any help and advice you can offer.

**Comment by sfischer13** [comment]
I managed to install Huginn on a RPi3 (Raspbian Stretch), but I had to replace `therubyracer` with `mini_racer` as suggested in this pull request #1961. @dsander, maybe this could be a reason to merge that PR?

**Comment by catalyst1987** [comment]
@sfischer13 Thanks for the response on this.  It is greatly appreciated.  Is switching to mini-racer something easy enough to do or would it require a code change?

**Comment by dsander** [comment]
@sfischer13 Good to hear that `mini_racer` now works on the PI, we can not merge the PR yet because Debian 7 is still supported by us and does not work with `mini_racer`.  LTS support for wheezy ends in June 2018.

**Comment by sfischer13** [comment]
If I remember correctly, I only applied the changes from this [this pull request](https://github.com/huginn/huginn/pull/1961/files). Yes, it's a code change, but it's only three small modifcations.

**Comment by catalyst1987** [comment]
@sfischer13 Yep, I did that.  And then when I run 'sudo -u huginn -H bundle install --deployment --without development test' I get the following error:

"You are trying to install in deployment mode after changing
your Gemfile. Run `bundle install` elsewhere and add the
updated Gemfile.lock to version control.

Bundler is unlocking ruby: (version
2.4.2
2.3.1)

You have added to the Gemfile:
* mini_racer (~> 0.1.9)  "

I'm sure I'm missing some kind of step here but am uncertain as of to what.

**Comment by dsander** [comment]
@catalyst1987 You also need to apply the changes from the `Gemfile.lock`

**Comment by catalyst1987** [comment]
Ah I understand now.  I did edit the Gemfile.lock but was missing a line.  I fixed that and it appeared to work but then I got this error:

"RUBY VERSION
   ruby 2.3.1p112

BUNDLED WITH
   1.15.4
```

--- TEMPLATE END ----------------------------------------------------------------

Unfortunately, an unexpected error occurred, and Bundler cannot continue.

First, try this link to see if there are any existing issue reports for this error:
https://github.com/bundler/bundler/search?q=Unable+to+find+a+spec+satisfying+therubyracer+%28%7E%3E+0.12.3%29+in+the+set.+Perhaps+the+lockfile+is+corrupted%3F&type=Issues

If there aren't any reports for this error yet, please create copy and paste the report template above into a new issue. Don't forget to anonymize any private data! The new issue form is located at:
https://github.com/bundler/bundler/issues/new
"

Do  I need to change the version on the "bundled with" on both the files or is this some other kind of problem?

Many Thanks.

**Comment by dsander** [comment]
It's possible that the `Gemfile.lock` still has a syntax error.

I updated the branch of my pull request to be ontop of master, you can check it out locally:

```
git fetch origin pull/1961/head:mini_racer
git checkout mini_racer
```

Then follow the normal upgrade procedure https://github.com/huginn/huginn/blob/master/doc/manual/update.md#4-install-gems-migrate-and-precompile-assets

**Comment by catalyst1987** [comment]
Okay, thanks.  Done all that and tried the export again, but it still seems to be stuck on Exporting Services

sudo bundle exec rake production:export --trace
** Invoke production:export (first_time)
** Invoke production:check (first_time)
** Execute production:check
** Execute production:export
** Execute production:stop
Stopping huginn ...
Exporting new services ...

**Comment by kraftzwerg** [comment]
Did you managed to get this working?
I am building an docker arm image for the RP4, and face the same Problem.

root@7008eecce5c4:/home/huginn/huginn# sudo bundle exec rake production:export
Stopping huginn ...
Exporting new services ...

and this "Exporting new services ..." never ends
